libs/generic-view/ui/src/lib/interactive/form/input/dynamic-input/dynamic-input-list.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import {
+  FormDynamicInputConfig,
+  FormDynamicInputData,
+} from "generic-view/models"
+import { APIFC } from "generic-view/utils"
+import React, { useEffect } from "react"
+import styled from "styled-components"
+import { DynamicInputRow } from "./dynamic-input-row"
+import { useFormContext } from "react-hook-form"
+
+export const DynamicInputList: APIFC<
+  FormDynamicInputData,
+  FormDynamicInputConfig
+> = ({ data, config }) => {
+  const { setValue, watch } = useFormContext()
+  // const formValues = watch(
+  //   data?.values.map((_, i) => `${config.name}-${i}-isDefault`) || []
+  // )
+
+  useEffect(() => {
+    data?.values.forEach((value, index) =>
+      setValue(`${config.name}-${index}-value`, value.value)
+    )
+  }, [data, setValue, config.name])
+
+  const handleSetDefault = (index: number) => {
+    const formValues = watch()
+
+    Object.entries(formValues).forEach(([key, _]) => {
+      if (key.endsWith("-isDefault")) {
+        setValue(key, false)
+      }
+    })
+
+    setValue(`${config.name}-${index}-isDefault`, true)
+  }
+
+  return (
+    <ListWrapper>
+      <Label>{config.label}</Label>
+      {data?.values.map((value, index) => (
+        <DynamicInputRow
+          key={index}
+          name={`${config.name}-${index}`}
+          options={config.options}
+          type={config.type}
+          data={value}
+          onSetDefault={() => handleSetDefault(index)}
+        />
+      ))}
+    </ListWrapper>
+  )
+}
+
+const ListWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 0.8rem;
+`
+
+const Label = styled.label`
```

Comment:
"What do you think about using the value from typographyConfig, e.g., typographyConfig["typography.p3"]?"

```
<Label
  {...typographyConfig["typography.p3"]}
>
  {config.label}
</Label>
```